### PR TITLE
feat: AB#69 Handle Preview image of apptemplate

### PR DESCRIPTION
**Pull request by DerCed**
## Description
Added apptemplate image preview

## Motivation and Context
Added image to better differentiate between apptemplates

## How has this been tested?
Added string image file to demo template
Checked regex in regex 101
A wrong specified image results in ``Template configuration could not be loaded. Please check config file (template-config.json).``

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: persistence/src/main/java/de/starwit/persistence/entity/AppTemplate.java [hunk 1]
```diff
@@ -1,126 +1,130 @@
...
+
+	@NotBlank
+	@Pattern(regexp = "^([a-zA-Z_0-9]|-)*$")
+	@Size(max = 100)
+	@Column(name = "PACKAGE", nullable = false, length = 100)
+	private String packagePlaceholder = "starwit";
+
+	@Column(name = "IMAGE_URL")
+	@Pattern(regexp = "^(https?:\\/\\/.*\\.(?:png|jpg|jpeg))$")
+	private String imageUrl;
```
Comment (DerCed) [inline persistence/src/main/java/de/starwit/persistence/entity/AppTemplate.java:32]: Änderung hier

File: service/src/main/java/de/starwit/dto/SaveAppTemplateDto.java [hunk 2]
```diff
@@ -1,89 +1,100 @@
...
+
+    private List<String> userGroups;
+
+    @Schema(hidden = true)
+    @Pattern(regexp = "^([a-zA-Z_0-9|-])*$")
+	@Size(max = 100)
+    private String packagePlaceholder;
+
+    @Size(max = 100)
+    private String imageUrl;
```
Comment (DerCed) [inline service/src/main/java/de/starwit/dto/SaveAppTemplateDto.java:37]: Änderung hier

File: service/src/main/java/de/starwit/mapper/AppTemplateMapper.java [hunk 3]
```diff
@@ -1,39 +1,41 @@
...
+        SaveAppTemplateDto dto = new SaveAppTemplateDto();
+        dto.setId(entity.getId());
+        dto.setName(entity.getTemplateName());
+        dto.setLocation(entity.getLocation());
+        dto.setBranch(entity.getBranch());
+        dto.setDescription(entity.getDescription());
+        dto.setCredentialsRequired(entity.isCredentialsRequired());
+        dto.setPackagePlaceholder(entity.getPackagePlaceholder());
+        dto.setGroups(entity.getGroups());
+        dto.setImageUrl(entity.getImageUrl());
```
Comment (DerCed) [inline service/src/main/java/de/starwit/mapper/AppTemplateMapper.java:22]: Änderung hier

File: service/src/main/java/de/starwit/mapper/AppTemplateMapper.java [hunk 4]
```diff
@@ -1,39 +1,41 @@
...
+        AppTemplate entity = new AppTemplate();
+        entity.setId(dto.getId());
+        entity.setTemplateName(dto.getName());
+        entity.setLocation(dto.getLocation());
+        entity.setBranch(dto.getBranch());
+        entity.setDescription(dto.getDescription());
+        entity.setCredentialsRequired(dto.isCredentialsRequired());
+        entity.setPackagePlaceholder(dto.getPackagePlaceholder());
+        entity.setGroups(dto.getGroups());
+        entity.setImageUrl(dto.getImageUrl());
```
Comment (DerCed) [inline service/src/main/java/de/starwit/mapper/AppTemplateMapper.java:37]: Änderung hier

File: service/src/main/java/de/starwit/mapper/AppTemplateMapper.java [hunk 5]
```diff
@@ -1,39 +1,41 @@
-package de.starwit.mapper;
```
Comment (witchpou) [inline service/src/main/java/de/starwit/mapper/AppTemplateMapper.java:1]: rejected because of formatting

**Review by witchpou — CHANGES_REQUESTED** [review]
undo formatting

File: service/src/main/java/de/starwit/mapper/AppTemplateMapper.java (hunk 5 above)
Comment (witchpou) [reply service/src/main/java/de/starwit/mapper/AppTemplateMapper.java:1]: line endings changed to CRLF
image

**Pull request merged**